TWTValidation/JSON Validator/Parser/TWTJSONSchemaParser.m

```diff
@@ -101,107 +110,75 @@ - (TWTJSONSchemaTopLevelASTNode *)parseWithError:(NSError *__autoreleasing *)out
 # pragma mark - Schema parser methods
 
 // Starting point for all schema nodes (except top level)
-// Directs the schema toward the appropriate parser method based on the value of "type", and guarantees that value is valid
-// Valid values for type are:
-//    A. Nil (type keyword not present)
-//    B. An array of strings that are all valid JSON types
-//    C. A string that is a valid JSON type
-// Type-specific parser methods are responsible for the entire process of creating a node from a schema, including:
-//     1. Initializing the appropriate ASTNode
-//     2. Setting the type (GenericASTNode only; all others are built into the node class)
-//     3. Parsing all common and type-specific properties
-//     4. Returning a completed node
+// Evaluates the "type" keyword, creates a node of the appropriate class, directs the schema toward the appropriate parser method, and parses common keywords
+// Type-specific parser methods are responsible for:
+//     1. Setting the type (GenericASTNode and AmbiguousASTNode only; all others are built into the node class)
+//     2. Parsing all type-specific properties
 - (TWTJSONSchemaASTNode *)parseSchema:(NSDictionary *)schema
 {
     NSParameterAssert(schema);
 
     [self failIfObject:schema isNotKindOfClass:[NSDictionary class] allowsNil:NO];
 
-    // Type is unspecified
-    NSString *type = schema[TWTJSONSchemaKeywordType];
-    if (!type) {
-        return [self parseGenericSchema:schema];
-    }
-
     [self pushPathComponent:TWTJSONSchemaKeywordType];
+    BOOL isTypeExplicit = NO;
+    id type = [self parseTypeKeywordForSchema:schema explicit:&isTypeExplicit];
+    id node = nil;
```

Comment:
This is where the casting previously was, I changed to using id and then initializing the actual object inside the if statements. I'm naming the variable `node` outside of these in order to set typeExplicit, parse common keywords, and return in one place (lines 150-152). Thoughts?